core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java

```diff
@@ -123,21 +126,13 @@ public static boolean isCharTypeComparable(
       SqlCallBinding binding,
       List<SqlNode> operands,
       boolean throwOnFailure) {
-    final SqlValidator validator = binding.getValidator();
-    final SqlValidatorScope scope = binding.getScope();
-    assert operands != null;
-    assert operands.size() >= 2;
+    requireNonNull(operands, "operands");
```

Comment:
Maybe we can set a descriptive/imperative message here, like `operands must not be null` or `operands is null`, it will be more clear?
Same for the call in other places.

Reply:
> Maybe we can set a descriptive/imperative message here, like operands must not be null or operands is null, it will be more clear?

The error message would be like `NullPointerException: operands`.
I guess it is more-or-less clear.

There should be a separate PR that replaces `requireNonNull(var);` with `requireNonNull(var, "var");` and `assert var != null;` with `requireNonNull(var, "var");`

Reply:
@vlsi ok, maybe it's clear enough, but I still prefer a descriptive/imperative message, anyway, it just doesn't matters.